src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -153,20 +153,24 @@ public Thread(ParameterizedThreadStart start, int maxStackSize)
             Initialize();
         }
 
-#if !TARGET_BROWSER
-        internal static bool IsThreadStartSupported => true;
-#else
-#if FEATURE_WASM_THREADS
+#if !TARGET_BROWSER || FEATURE_WASM_THREADS
         internal static bool IsThreadStartSupported => true;
+        internal static bool IsInternalThreadStartSupported => true;
+#elif FEATURE_WASM_PERFTRACING
```

Comment:
Does this have any relationship with the wider, FEATURE_PERFTRACING used to guard all perftracing support?

Reply:
I think `FEATURE_PERFTRACING` can be true or false on the wasm threaded runtime, ie it's the general perftracing support. Whereas `FEATURE_WASM_PERFTRACING` is the specific configuration where user threading is disabled.

Reply:
Yes, FEATURE_PERFTRACING will turn of all diagnostics supports, both in BCL and runtime. So can you build EventPipe/Diagnostic support on WASM without user threading enabled and get something to work?